### PR TITLE
TPM1.2: implement GetPermanentFlags

### PR DESCRIPTION
**Pull request by ChriMarMe**

### CODE REVIEW

**Comment by ChriMarMe** [comment]
@tmroeder 
aaaannnnd here. 

Thank you in advance :)

**Comment by ChriMarMe** [comment]
Really need some review here please

File: tpm/structures.go [hunk 1]
```diff
@@ -84,6 +84,31 @@ type capVersionInfoFixed struct {
 	VendorID  byte
 }
 
+// PermanentFlags contains a persistent TPM properties
```
Comment (awly) [inline tpm/structures.go]: "// PermanentFlags contains persistent TPM properties."

File: tpm/structures.go [hunk 2]
```diff
@@ -84,6 +84,31 @@ type capVersionInfoFixed struct {
...
+type PermanentFlags struct {
+	Tag                          uint16
+	Disable                      bool
+	Ownership                    bool
+	Deactivated                  bool
+	ReadPubek                    bool
```
Comment (awly) [inline tpm/structures.go]: ReadPubEK https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: tpm/structures.go [hunk 3]
```diff
@@ -84,6 +84,31 @@ type capVersionInfoFixed struct {
...
+	DisableOwnerClear            bool
+	AllowMaintenance             bool
+	PhysicalPresenceLifetimeLock bool
+	PhysicalPresenceHWEnable     bool
+	PhysicalPresenceCMDEnable    bool
+	CEKPUsed                     bool
+	TPMpost                      bool
```
Comment (awly) [inline tpm/structures.go]: TPMPost and TPMPostLock

File: tpm/structures.go [hunk 4]
```diff
@@ -84,6 +84,31 @@ type capVersionInfoFixed struct {
...
+	TPMpostLock                  bool
+	FIPS                         bool
+	Operator                     bool
+	EnableRevokeEK               bool
+	NvLocked                     bool
```
Comment (awly) [inline tpm/structures.go]: NVLocked

File: tpm/structures.go [hunk 5]
```diff
@@ -84,6 +84,31 @@ type capVersionInfoFixed struct {
...
+	ReadSRKPub                   bool
+	TpmEstablished               bool
```
Comment (awly) [inline tpm/structures.go]: TPMEstablished

File: tpm/tpm.go [hunk 6]
```diff
@@ -975,6 +975,19 @@ func GetManufacturer(rw io.ReadWriter) ([]byte, error) {
 	return getCapability(rw, capProperty, tpmCapPropManufacturer)
 }
 
+// GetPermanentFlags returns the TPM_CAP_FLAG_PERMANENT structure
```
Comment (awly) [inline tpm/tpm.go]: ... the TPM_PERMANENT_FLAGS structure.

**Pull request merged**